World and dynamic lighting

computer-kyle opened the pull request:
## Description

Makes the world darker and adds dynamic lighting (overhead florescent lights). The lights react in real time to the antagonist.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [x] Enhancement
- [ ] Refactor
- [x] Documentation
- [ ] Other (please specify):

## Related Issue

N / A

## Changes Made

- Adds real-time dynamic lighting to the world, and most world-building materials
- Adds overhead florescent light objects
- Adds dynamic light sources to the overhead lights, controlled by a script. This lighting turns off and on, with a flickering effect and material modulation for realism.
- Long flickers indicate that the antagonist is very near to the light
- Short flickers happen at random intervals

## Checklist

- [x] Code follows the project's coding style.
- [x] If asset(s) added, they have been made into a prefab
- [ ] Tests have been added/updated.
- [x] Documentation has been updated (if applicable).
- [x] No unused code or assets remain.
- [x] Unity project builds without errors.

## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/dbc0db7d-64cc-4c26-a39a-29a26a9b5993)

## Additional Notes

This update does seem to have a nonzero performance impact. Please report back on how much the extent of this is. To me, it seems to be about 20% of framerate. If the performance of this update impacts your own development progress, feel free to disable all lights using the ENABLED boolean in CeilingLightLogic

## Relevant UML Diagrams
![image](https://github.com/user-attachments/assets/40cc14d0-ddda-4066-978a-2bed74e30a91)

computer-kyle committed:
Material Lighting Update

- Changes material type from "Unlit/Texture" to "Universal Render Pipeline/Lit"

- Modified global lighting settings to achieve a darker map, making it feel more scary

computer-kyle committed:
Added ceiling lamp object

- Added a 3D model of a lamp that will go on the ceilings of the map and act as a dynamic light source

computer-kyle committed:
Updated ceiling light and added script

- Changed ceiling light asset type to fbx, which supports two materials in the object (fixture and light bulb)

- Added a script for each instance of a light that will react to the antagonist by flickering off and back on, depending on the range

computer-kyle committed:
Turned the lamp into a prefab

- Prefabbed lamp for modularity
- Removed once instance in Scene

computer-kyle committed:
Added comments to dynamic ceiling lamp script

computer-kyle committed:
Updated Materials

- Updated materials so they are no longer packaged with the object

- Changed code to modify the intensity of the lights, which are variable when it is flickering to look more realistic

computer-kyle committed:
Added more lights throughout the map

computer-kyle committed:
Added feature to disable all lights

- Added a boolean in CeilingLogic that allows for all lights to be disabled for performance and debugging

computer-kyle committed:
Added more lamps :p

- Most of the map has instances of lamps. I think it's good enough at least

computer-kyle committed:
Added doors to Map Core prefab, removed from SampleScene

uid106 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - Adds overhead lighting
  - Darkens assets on the map
  - Creates scripting to change lighting as the antagonist passes through the map.

- **Unused Code**: 
  - None

- **Code Style**: 
   - Consistent with project expectations

- **Script Structure**: 
   -  Structure of script is proper for this project

- **Documentation**: 
   - Well documented

- **Build Integrity**: 
  - Project compiles, uses more GPU but I think this is a side effect of not being a full build that is frame limited, my frames don't suffer significantly. 

---

### Summary
The changes to the map assets make the map look significantly better, more like a horror map now that it can be dark. The lighting functions very well with the ghost as he passes through the map. This addition looks ready to merge.

computer-kyle committed:
Updated CeilingLampLogic.cs to include random flickering

- Added a random event for each individual light to flicker. This flicker is a once-off-once-on action, and it is visually different from the flicker that occurs when the antagonist nears a lamp.

krain0 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - **Real-time dynamic lighting:** The new lighting seems to work as intended, and I think it greatly improves the look of the game.
  - **Florescent light:** This asset fits in well with the map.
  - **Flickering lights:** The flickering lights also seem to work as intended, and impove the ambience of the map.

- **Unused Code**:
  - None

- **Code Style**:
  - Everything looks good and matches our project's style.

- **Script Structure**:
  - Well-organized, I don't see any problems.

- **Documentation**:
  - The comments clearly describe the intent of your code.

- **Build Integrity**:
  - The game built without issue on my machine.

---

### Summary
All of the changes you made to the lighting look great. I think making the map darker was a great idea, and the flickering when the antagonist is close makes it seem spookier. Everything looks well-implemented. My frame-rate is lower, but still acceptable, and I think this was likely inevitable with this sort of change. I think this PR is ready to be merged!
